File: commands/digitalungdom/bug.js
/* global db include guild */

const getUserByDiscordId = include( 'models/get' ).getUserByDiscordId;
const createNotificationEmbed = include( 'utils/embeds/createNotificationEmbed' );

module.exports = {
  name: 'bug',
  description: 'Skicka en bug report till Digital Ungdom.',
  aliases: [ 'bugg' ],
  group: 'digitalungdom',
  usage: 'bug <bugg>',
  example: 'bug verifierings funktionen funkar inte',
  serverOnly: false,
  adminOnly: false,
  async execute( message, args ) {
    if ( args.length === 0 ) return message.reply( 'Du måste skicka med ett kort meddelande.' );
    const bug = args.join( ' ' );
    const authorID = message.author.id;
    const authorUsername = message.author.username;
    let id = await getUserByDiscordId( authorID );
    if ( id ) id = id._id;

    await db.collection( 'notifications' ).insertOne( {
      'type': 'bug',
      'where': 'discord',
      'message': bug,
      'author': {
        'id': id,
        'discordID': authorID,
        'discordUsername': authorUsername
      }
    } );

    const notification = createNotificationEmbed( 'BUG', bug, 16711680, { 'id': authorID, 'name': authorUsername, 'url': message.author.displayAvatarURL } );
    const notificationChannel = guild.channels.find( ch => ch.name === 'notifications' );
    notificationChannel.send( '@here', { 'embed': notification } );

    return message.reply( 'tack för din medverkan!' );
  },
};

File: commands/digitalungdom/bug.go
package digitalungdom

import (
	"strings"

	"github.com/digitalungdom-se/dub/internal"
	"github.com/digitalungdom-se/dub/pkg"
	"go.mongodb.org/mongo-driver/bson"
)

var Bug = internal.Command{
	Name:        "bug",
	Description: "Anmäl ett bugg",
	Aliases:     []string{"bugg"},
	Group:       "digitalungdom",
	Usage:       "bug <message>",
	Example:     "bug boten spelar inte musik",
	ServerOnly:  false,
	AdminOnly:   false,

	Execute: func(ctx *pkg.Context, server *internal.Server) error {
		if len(ctx.Args) < 3 {
			_, err := ctx.Reply("Du måste ge en anledning på minst tre ord.")
			return err
		}

		bug := bson.M{
			"type":    "bug",
			"where":   "discord",
			"message": strings.Join(ctx.Args, " "),
			"author":  ctx.Message.Author.ID}

		err := server.Database.InsertNotification(bug)
		if err != nil {
			return err
		}

		_, err = ctx.Reply("Du har nu anmält denna bugg till Digital Ungdom. Tack för din medverkan!")
		if err != nil {
			return err
		}

		return nil
	},
}
